contrib/inventory/foreman.py

```diff
@@ -158,7 +182,7 @@ def _get_json(self, url, ignore_errors=None):
                 return json['results']
             # List of all hosts is returned paginaged
             results = results + json['results']
-            if len(results) >= json['total']:
+            if len(results) >= json['subtotal']:
```

Comment:
When you query the API with a search filter, "total" still gives you the total number of objects (hosts for example) that you would get without the search filter.

"subtotal" gives you the number of objects that are returned with the search filter applied. If there was no search filter, it's equal to "total". Without this change, you get this warning: "Did not make any progress during loop."

https://access.redhat.com/documentation/en-us/red_hat_satellite/6.2/html-single/api_guide/
total — The total number of objects without any search parameters.
subtotal — The number of objects returned with the given search parameters (if there is no search, then subtotal is equal to total).